objects/src/accounts/seed.rs

```diff
@@ -91,7 +93,7 @@ pub fn get_account_seed_inner(
 
         // regularly check if another thread found a digest
         count += 1;
-        if count % 500_000 == 0 && stop.get().is_some() {
+        if count % 500_000 == 0 && *stop.read().unwrap() {
```

Comment:
The guard is dropped after the `if` condition is evaluated [ref](https://doc.rust-lang.org/reference/destructors.html?highlight=drop#temporary-scopes)